apps/ensrainbow/src/index.ts

```diff
@@ -0,0 +1,41 @@
+import { Hono } from 'hono';
+import { serve } from '@hono/node-server';
+import { ClassicLevel } from 'classic-level';
+import { join } from 'path';
+import type { Context } from 'hono';
+
+const app = new Hono();
+const DATA_DIR = process.env.DATA_DIR || join(process.cwd(), 'data');
+
+// Initialize LevelDB with proper types for key and value
+const db = new ClassicLevel<Buffer, string>(DATA_DIR, {
+    valueEncoding: 'utf8',
+    keyEncoding: 'binary'
+});
+
+app.get('/v1/heal/:labelhash', async (c: Context) => {
+    const labelhash = c.req.param('labelhash');
+    
+    try {
+        // Convert hex string to Buffer, stripping '0x' prefix if present
+        const hashBytes = Buffer.from(labelhash.replace(/^0x/, ''), 'hex');
+        const label = await db.get(hashBytes);
+        return c.json({ healed: label });
```

Comment:
Perhaps we can just return the text value, as it's the only possible value to be returned:
```suggestion
        return c.text(label);
```